src/GridFS/Bucket.php

```diff
@@ -0,0 +1,367 @@
+        $this->collectionsWrapper->getFilesCollection()->deleteOne(['_id' => $id]);
+        $this->collectionsWrapper->getChunksCollection()->deleteMany(['files_id' => $id]);
+
+        if ($file === null) {
+            throw new GridFSFileNotFoundException($id, $this->collectionsWrapper->getFilesCollection()->getNameSpace());
+        }
+
+    }
+
+    /**
+     * Writes the contents of a GridFS file to a writable stream.
+     *
+     * @param ObjectId $id          ObjectId of the file
+     * @param resource $destination Writable Stream
+     * @throws GridFSFileNotFoundException
+     */
+    public function downloadToStream(ObjectId $id, $destination)
+    {
+        $file = $this->collectionsWrapper->getFilesCollection()->findOne(
+            ['_id' => $id],
+            ['typeMap' => ['root' => 'stdClass']]
+        );
+
+        if ($file === null) {
+            throw new GridFSFileNotFoundException($id, $this->collectionsWrapper->getFilesCollection()->getNameSpace());
+        }
+
+        $gridFsStream = new GridFSDownload($this->collectionsWrapper, $file);
+        $gridFsStream->downloadToStream($destination);
+    }
+
+    /**
+     * Writes the contents of a GridFS file, which is selected by name and
+     * revision, to a writable stream.
+     *
+     * Supported options:
+     *
+     *  * revision (integer): Which revision (i.e. documents with the same
+     *    filename and different uploadDate) of the file to retrieve. Defaults
+     *    to -1 (i.e. the most recent revision).
+     *
+     * Revision numbers are defined as follows:
+     *
+     *  * 0 = the original stored file
+     *  * 1 = the first revision
+     *  * 2 = the second revision
+     *  * etc…
+     *  * -2 = the second most recent revision
+     *  * -1 = the most recent revision
+     *
+     * @param string   $filename    File name
+     * @param resource $destination Writable Stream
+     * @param array    $options     Download options
+     * @throws GridFSFileNotFoundException
+     */
+    public function downloadToStreamByName($filename, $destination, array $options = [])
+    {
+        $options += ['revision' => -1];
+        $file = $this->findFileRevision($filename, $options['revision']);
+        $gridFsStream = new GridFSDownload($this->collectionsWrapper, $file);
+        $gridFsStream->downloadToStream($destination);
+    }
+
+    /**
+    * Drops the files and chunks collection associated with GridFS this bucket
+    *
+    */
+
+    public function drop()
+    {
+        $this->collectionsWrapper->dropCollections();
+    }
+
+    /**
+     * Find files from the GridFS bucket's files collection.
+     *
+     * @see Find::__construct() for supported options
+     * @param array|object $filter  Query by which to filter documents
+     * @param array        $options Additional options
+     * @return Cursor
+     */
+    public function find($filter, array $options = [])
```

Comment:
Per https://github.com/mongodb/mongo-php-library/pull/57#commitcomment-15191195, I think we should default `$filter` to an empty array.